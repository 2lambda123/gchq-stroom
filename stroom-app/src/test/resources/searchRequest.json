{
  "key": {
    "uuid": "3|Test Dashboard|query-MRGPM|VW3P_1490104737989"
  },
  "query": {
    "dataSource": {
      "type": "Index",
      "uuid": "57a35b9a-083c-4a93-a813-fc3ddfe1ff44",
      "name": "Example index"
    },
    "expression": {
      "type": "operator",
      "op": "AND",
      "children": [
        {
          "type": "term",
          "field": "EventTime",
          "condition": "BETWEEN",
          "value": "2010-01-01T00:00:00.000Z,2010-01-01T00:10:00.000Z",
          "enabled": true
        }
      ],
      "enabled": true
    }
  },
  "resultRequests": [
    {
      "componentId": "vis-L1AL1",
      "mappings": [
        {
          "queryId": "query-MRGPM",
          "fields": [
            {
              "name": "EventTime",
              "expression": "roundMinute(${EventTime})",
              "format": {
                "type": "DATE_TIME"
              },
              "group": 0
            },
            {
              "name": "UserId",
              "expression": "${UserId}",
              "format": {
                "type": "GENERAL"
              },
              "group": 0
            },
            {
              "name": "Count",
              "expression": "count()",
              "format": {
                "type": "NUMBER"
              }
            },
            {
              "name": "StreamId",
              "expression": "${StreamId}"
            },
            {
              "name": "EventId",
              "expression": "${EventId}"
            }
          ],
          "extractValues": true,
          "extractionPipeline": {
            "type": "Pipeline",
            "uuid": "e5ecdf93-d433-45ac-b14a-1f77f16ae4f7",
            "name": "Example extraction"
          },
          "maxResults": [
            1000000
          ]
        },
        {
          "fields": [
            {
              "expression": "${EventTime}",
              "sort": {
                "direction": "ASCENDING"
              },
              "format": {
                "type": "DATE_TIME"
              },
              "group": 0
            },
            {
              "expression": "${UserId}",
              "sort": {
                "direction": "ASCENDING"
              },
              "format": {
                "type": "GENERAL"
              },
              "group": 1
            },
            {
              "expression": "${UserId}",
              "sort": {
                "direction": "ASCENDING"
              },
              "format": {
                "type": "GENERAL"
              }
            },
            {
              "expression": "${Count}",
              "format": {
                "type": "NUMBER"
              }
            }
          ],
          "maxResults": [
            20,
            10,
            null
          ],
          "showDetail": true
        }
      ],
      "resultStyle": "FLAT",
      "fetch": "ALL"
    },
    {
      "componentId": "vis-SPSCW",
      "mappings": [
        {
          "queryId": "query-MRGPM",
          "fields": [
            {
              "name": "EventTime",
              "expression": "roundMinute(${EventTime})",
              "format": {
                "type": "DATE_TIME"
              },
              "group": 0
            },
            {
              "name": "UserId",
              "expression": "${UserId}",
              "format": {
                "type": "GENERAL"
              },
              "group": 0
            },
            {
              "name": "Count",
              "expression": "count()",
              "format": {
                "type": "NUMBER"
              }
            },
            {
              "name": "StreamId",
              "expression": "${StreamId}"
            },
            {
              "name": "EventId",
              "expression": "${EventId}"
            }
          ],
          "extractValues": true,
          "extractionPipeline": {
            "type": "Pipeline",
            "uuid": "e5ecdf93-d433-45ac-b14a-1f77f16ae4f7",
            "name": "Example extraction"
          },
          "maxResults": [
            1000000
          ]
        },
        {
          "fields": [
            {
              "sort": {
                "direction": "ASCENDING"
              },
              "format": {
                "type": "GENERAL"
              },
              "group": 0
            },
            {
              "expression": "${UserId}",
              "sort": {
                "direction": "ASCENDING"
              },
              "format": {
                "type": "GENERAL"
              },
              "group": 1
            },
            {
              "expression": "${EventTime}",
              "sort": {
                "direction": "ASCENDING"
              },
              "format": {
                "type": "DATE_TIME"
              }
            },
            {
              "expression": "${Count}",
              "format": {
                "type": "NUMBER"
              }
            }
          ],
          "maxResults": [
            20,
            100,
            null
          ],
          "showDetail": true
        }
      ],
      "resultStyle": "FLAT",
      "fetch": "ALL"
    },
    {
      "componentId": "table-78LF4",
      "mappings": [
        {
          "queryId": "query-MRGPM",
          "fields": [
            {
              "name": "EventTime",
              "expression": "roundMinute(${EventTime})",
              "format": {
                "type": "DATE_TIME"
              },
              "group": 0
            },
            {
              "name": "UserId",
              "expression": "${UserId}",
              "format": {
                "type": "GENERAL"
              },
              "group": 0
            },
            {
              "name": "Count",
              "expression": "count()",
              "format": {
                "type": "NUMBER"
              }
            },
            {
              "name": "StreamId",
              "expression": "${StreamId}"
            },
            {
              "name": "EventId",
              "expression": "${EventId}"
            }
          ],
          "extractValues": true,
          "extractionPipeline": {
            "type": "Pipeline",
            "uuid": "e5ecdf93-d433-45ac-b14a-1f77f16ae4f7",
            "name": "Example extraction"
          },
          "maxResults": [
            1000000
          ]
        }
      ],
      "requestedRange": {
        "offset": 0,
        "length": 100
      },
      "resultStyle": "TABLE",
      "fetch": "ALL"
    },
    {
      "componentId": "table-BKJT6",
      "mappings": [
        {
          "queryId": "query-MRGPM",
          "fields": [
            {
              "name": "UserId",
              "expression": "${UserId}",
              "format": {
                "type": "GENERAL"
              },
              "group": 0
            },
            {
              "name": "Count",
              "expression": "count()",
              "format": {
                "type": "NUMBER"
              }
            },
            {
              "name": "StreamId",
              "expression": "${StreamId}"
            },
            {
              "name": "EventId",
              "expression": "${EventId}"
            }
          ],
          "extractValues": true,
          "extractionPipeline": {
            "type": "Pipeline",
            "uuid": "e5ecdf93-d433-45ac-b14a-1f77f16ae4f7",
            "name": "Example extraction"
          },
          "maxResults": [
            1000000
          ]
        }
      ],
      "requestedRange": {
        "offset": 0,
        "length": 100
      },
      "resultStyle": "TABLE",
      "fetch": "ALL"
    },
    {
      "componentId": "vis-QYG7H",
      "mappings": [
        {
          "queryId": "query-MRGPM",
          "fields": [
            {
              "name": "UserId",
              "expression": "${UserId}",
              "format": {
                "type": "GENERAL"
              },
              "group": 0
            },
            {
              "name": "Count",
              "expression": "count()",
              "format": {
                "type": "NUMBER"
              }
            },
            {
              "name": "StreamId",
              "expression": "${StreamId}"
            },
            {
              "name": "EventId",
              "expression": "${EventId}"
            }
          ],
          "extractValues": true,
          "extractionPipeline": {
            "type": "Pipeline",
            "uuid": "e5ecdf93-d433-45ac-b14a-1f77f16ae4f7",
            "name": "Example extraction"
          },
          "maxResults": [
            1000000
          ]
        },
        {
          "fields": [
            {
              "sort": {
                "direction": "ASCENDING"
              },
              "format": {
                "type": "GENERAL"
              },
              "group": 0
            },
            {
              "expression": "${UserId}",
              "format": {
                "type": "GENERAL"
              }
            },
            {
              "expression": "${Count}",
              "format": {
                "type": "NUMBER"
              }
            }
          ],
          "maxResults": [
            20,
            20
          ],
          "showDetail": true
        }
      ],
      "resultStyle": "FLAT",
      "fetch": "ALL"
    }
  ],
  "dateTimeLocale": "Etc/GMT",
  "incremental": false
}