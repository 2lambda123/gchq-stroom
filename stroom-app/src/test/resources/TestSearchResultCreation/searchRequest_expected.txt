{
  "key" : {
    "uuid" : "e177cf16-da6c-4c7d-a19c-09a201f5a2da|Test Dashboard|query-MRGPM|CO9U_1605714426717"
  },
  "query" : {
    "dataSource" : {
      "type" : "Index",
      "uuid" : "57a35b9a-083c-4a93-a813-fc3ddfe1ff44",
      "name" : "Example index"
    },
    "expression" : {
      "type" : "operator",
      "children" : [ {
        "type" : "term",
        "field" : "EventTime",
        "condition" : "BETWEEN",
        "value" : "2010-01-01T00:00:00.000Z,2010-01-01T00:10:00.000Z"
      } ]
    },
    "params" : [ {
      "key" : "currentUser()",
      "value" : "admin"
    } ]
  },
  "resultRequests" : [ {
    "componentId" : "table-BKJT6",
    "mappings" : [ {
      "queryId" : "query-MRGPM",
      "fields" : [ {
        "id" : "table-BKJT6|TRGIL",
        "name" : "UserId",
        "expression" : "${UserId}",
        "sort" : {
          "order" : 0,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "GENERAL"
        },
        "group" : 0
      }, {
        "id" : "table-BKJT6|CER2D",
        "name" : "Count",
        "expression" : "count()",
        "format" : {
          "type" : "NUMBER"
        }
      }, {
        "id" : "__stream_id__",
        "name" : "__stream_id__",
        "expression" : "${StreamId}"
      }, {
        "id" : "__event_id__",
        "name" : "__event_id__",
        "expression" : "${EventId}"
      } ],
      "extractValues" : true,
      "extractionPipeline" : {
        "type" : "Pipeline",
        "uuid" : "e5ecdf93-d433-45ac-b14a-1f77f16ae4f7",
        "name" : "Example extraction"
      },
      "maxResults" : [ 1000000 ]
    } ],
    "requestedRange" : {
      "offset" : 0,
      "length" : 100
    },
    "resultStyle" : "TABLE",
    "fetch" : "CHANGES"
  }, {
    "componentId" : "vis-QYG7H",
    "mappings" : [ {
      "queryId" : "query-MRGPM",
      "fields" : [ {
        "id" : "table-BKJT6|TRGIL",
        "name" : "UserId",
        "expression" : "${UserId}",
        "sort" : {
          "order" : 0,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "GENERAL"
        },
        "group" : 0
      }, {
        "id" : "table-BKJT6|CER2D",
        "name" : "Count",
        "expression" : "count()",
        "format" : {
          "type" : "NUMBER"
        }
      }, {
        "id" : "__stream_id__",
        "name" : "__stream_id__",
        "expression" : "${StreamId}"
      }, {
        "id" : "__event_id__",
        "name" : "__event_id__",
        "expression" : "${EventId}"
      } ],
      "extractValues" : true,
      "extractionPipeline" : {
        "type" : "Pipeline",
        "uuid" : "e5ecdf93-d433-45ac-b14a-1f77f16ae4f7",
        "name" : "Example extraction"
      },
      "maxResults" : [ 1000000 ]
    }, {
      "fields" : [ {
        "sort" : {
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "GENERAL"
        },
        "group" : 0
      }, {
        "expression" : "${UserId}",
        "format" : {
          "type" : "GENERAL"
        }
      }, {
        "expression" : "${Count}",
        "format" : {
          "type" : "NUMBER"
        }
      } ],
      "maxResults" : [ 20, 20 ],
      "showDetail" : true
    } ],
    "resultStyle" : "FLAT",
    "fetch" : "CHANGES"
  }, {
    "componentId" : "table-78LF4",
    "mappings" : [ {
      "queryId" : "query-MRGPM",
      "fields" : [ {
        "id" : "table-78LF4|J8IX8",
        "name" : "EventTime",
        "expression" : "roundMinute(${EventTime})",
        "sort" : {
          "order" : 0,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "DATE_TIME"
        },
        "group" : 0
      }, {
        "id" : "table-78LF4|FZ798",
        "name" : "UserId",
        "expression" : "${UserId}",
        "sort" : {
          "order" : 1,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "GENERAL"
        },
        "group" : 0
      }, {
        "id" : "table-78LF4|0TR7Y",
        "name" : "Count",
        "expression" : "count()",
        "format" : {
          "type" : "NUMBER"
        }
      }, {
        "id" : "__stream_id__",
        "name" : "__stream_id__",
        "expression" : "${StreamId}"
      }, {
        "id" : "__event_id__",
        "name" : "__event_id__",
        "expression" : "${EventId}"
      } ],
      "extractValues" : true,
      "extractionPipeline" : {
        "type" : "Pipeline",
        "uuid" : "e5ecdf93-d433-45ac-b14a-1f77f16ae4f7",
        "name" : "Example extraction"
      },
      "maxResults" : [ 1000000 ]
    } ],
    "requestedRange" : {
      "offset" : 0,
      "length" : 100
    },
    "resultStyle" : "TABLE",
    "fetch" : "CHANGES"
  }, {
    "componentId" : "vis-L1AL1",
    "mappings" : [ {
      "queryId" : "query-MRGPM",
      "fields" : [ {
        "id" : "table-78LF4|J8IX8",
        "name" : "EventTime",
        "expression" : "roundMinute(${EventTime})",
        "sort" : {
          "order" : 0,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "DATE_TIME"
        },
        "group" : 0
      }, {
        "id" : "table-78LF4|FZ798",
        "name" : "UserId",
        "expression" : "${UserId}",
        "sort" : {
          "order" : 1,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "GENERAL"
        },
        "group" : 0
      }, {
        "id" : "table-78LF4|0TR7Y",
        "name" : "Count",
        "expression" : "count()",
        "format" : {
          "type" : "NUMBER"
        }
      }, {
        "id" : "__stream_id__",
        "name" : "__stream_id__",
        "expression" : "${StreamId}"
      }, {
        "id" : "__event_id__",
        "name" : "__event_id__",
        "expression" : "${EventId}"
      } ],
      "extractValues" : true,
      "extractionPipeline" : {
        "type" : "Pipeline",
        "uuid" : "e5ecdf93-d433-45ac-b14a-1f77f16ae4f7",
        "name" : "Example extraction"
      },
      "maxResults" : [ 1000000 ]
    }, {
      "fields" : [ {
        "expression" : "${EventTime}",
        "sort" : {
          "order" : 0,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "DATE_TIME"
        },
        "group" : 0
      }, {
        "expression" : "${UserId}",
        "sort" : {
          "order" : 1,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "GENERAL"
        },
        "group" : 1
      }, {
        "expression" : "${UserId}",
        "sort" : {
          "order" : 2,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "GENERAL"
        }
      }, {
        "expression" : "${Count}",
        "format" : {
          "type" : "NUMBER"
        }
      } ],
      "maxResults" : [ 20, 10, 500 ],
      "showDetail" : true
    } ],
    "resultStyle" : "FLAT",
    "fetch" : "CHANGES"
  }, {
    "componentId" : "vis-SPSCW",
    "mappings" : [ {
      "queryId" : "query-MRGPM",
      "fields" : [ {
        "id" : "table-78LF4|J8IX8",
        "name" : "EventTime",
        "expression" : "roundMinute(${EventTime})",
        "sort" : {
          "order" : 0,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "DATE_TIME"
        },
        "group" : 0
      }, {
        "id" : "table-78LF4|FZ798",
        "name" : "UserId",
        "expression" : "${UserId}",
        "sort" : {
          "order" : 1,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "GENERAL"
        },
        "group" : 0
      }, {
        "id" : "table-78LF4|0TR7Y",
        "name" : "Count",
        "expression" : "count()",
        "format" : {
          "type" : "NUMBER"
        }
      }, {
        "id" : "__stream_id__",
        "name" : "__stream_id__",
        "expression" : "${StreamId}"
      }, {
        "id" : "__event_id__",
        "name" : "__event_id__",
        "expression" : "${EventId}"
      } ],
      "extractValues" : true,
      "extractionPipeline" : {
        "type" : "Pipeline",
        "uuid" : "e5ecdf93-d433-45ac-b14a-1f77f16ae4f7",
        "name" : "Example extraction"
      },
      "maxResults" : [ 1000000 ]
    }, {
      "fields" : [ {
        "sort" : {
          "order" : 0,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "GENERAL"
        },
        "group" : 0
      }, {
        "expression" : "${UserId}",
        "format" : {
          "type" : "GENERAL"
        },
        "group" : 1
      }, {
        "expression" : "${EventTime}",
        "sort" : {
          "order" : 2,
          "direction" : "ASCENDING"
        },
        "format" : {
          "type" : "DATE_TIME"
        }
      }, {
        "expression" : "${Count}",
        "format" : {
          "type" : "NUMBER"
        }
      } ],
      "maxResults" : [ 20, 100, 1000 ],
      "showDetail" : true
    } ],
    "resultStyle" : "FLAT",
    "fetch" : "CHANGES"
  } ],
  "dateTimeLocale" : "Europe/London",
  "incremental" : true
}